src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -481,5 +483,12 @@ private byte GetMappedAddress(Register register, Port port = Port.PortA, BankSty
 
         private static string GetBits(byte value) => Convert.ToString(value, 2).PadLeft(8, '0');
         private static string GetBits(ushort value) => Convert.ToString(value, 2).PadLeft(16, '0');
+
+        public void OpenPin(int pinNumber, PinMode mode) => SetPinMode(pinNumber, mode);
```

Comment:
NIT: GpioController throws an exception when trying to open a pin that is already open, same goes for trying to close, read or write to a pin that is closed, I wonder if we should match that behavior. Not saying we have to, I'm just thinking out loud here.

Reply:
Probably not as there is no such thing as an open/closed pin for this device. I don't think it would be worth sticking an additional state array in this to match something that doesn't technically matter. That said, do we validate mode when writing/reading?

Reply:
We do, and we also throw in that case.

https://github.com/dotnet/iot/blob/f56cbdaf0d1d4a181d4d6fb97f096e990fc0ce28/src/System.Device.Gpio/System/Device/Gpio/GpioController.cs#L164-L174

Reply:
That part I think we should add.